src/swap/saga.ts

```diff
@@ -0,0 +1,116 @@
+import { CeloTx } from '@celo/connect'
+import { TxParamsNormalizer } from '@celo/connect/lib/utils/tx-params-normalizer'
+import { ContractKit } from '@celo/contractkit'
+import { PayloadAction } from '@reduxjs/toolkit'
+import { call, put, select, takeLatest } from 'redux-saga/effects'
+import { maxSwapSlippagePercentageSelector } from 'src/app/selectors'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import {
+  swapApprove,
+  swapError,
+  swapExecute,
+  swapPriceChange,
+  swapStart,
+  swapSuccess,
+} from 'src/swap/slice'
+import { Field, SwapInfo, SwapTransaction } from 'src/swap/types'
+import { sendTransaction } from 'src/transactions/send'
+import { newTransactionContext } from 'src/transactions/types'
+import { fetchWithTimeout } from 'src/utils/fetchWithTimeout'
+import Logger from 'src/utils/Logger'
+import { getContractKit } from 'src/web3/contracts'
+import networkConfig from 'src/web3/networkConfig'
+import { getConnectedUnlockedAccount } from 'src/web3/saga'
+import { applyChainIdWorkaround, buildTxo } from 'src/web3/utils'
+
+const TAG = 'swap/saga'
+
+function getPriceDiff(price1: number, price2: number) {
+  return (Math.abs(price1 - price2) / ((price1 + price2) / 2)) * 100
+}
+
+export function* swapSubmitSaga(action: PayloadAction<SwapInfo>) {
```

Comment:
did we consider making this saga a hook? the benefit would be that the transient states (like the swap is being started / approved / executed / priced changed) would not need to be in redux since they can directly be displayed by the only component consuming this data. i don't have a strong opinion though, what is here works. just wanted to throw the idea out there, since we've been talking about our redux usage recently